v2/pkg/operators/extractors/compile.go

```diff
@@ -48,7 +47,7 @@ func (e *Extractor) CompileExtractors() error {
 	}
 
 	if e.CaseInsensitive {
-		if e.Type != "kval" {
+		if e.GetType().String() != "kval" {
```

Comment:
It should rather be `e.GetType() != KValExtractor`. The point is not to rely on hard-coded string comparisons

Reply:
Done